### PR TITLE
Always generate RSA keys of expected bit length

### PR DESCRIPTION
**Pull request by larabr**
- Fix RSA key generation code used when no native crypto library is available (i.e. no NodeCrypto or WebCrypto). Now generated keys are always of exact bit length. This was not guaranteed before, and it was common for keys to be one bit shorter than expected.
- Remove leftover code related to legacy WebCrypto interfaces (IE11 and Safari 10)

### CODE REVIEW

File: src/crypto/public_key/prime.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ import util from '../../util';
 import { getRandomBigInteger } from '../random';
 
 /**
- * Probabilistic random number generator
+ * Probabilistic prime random number generator
```
Comment (twiss) [inline src/crypto/public_key/prime.js:28]:
```suggestion
 * Probabilistic random prime number generator
```

File: src/crypto/public_key/rsa.js [hunk 2]
```diff
@@ -178,41 +163,20 @@ export async function generate(bits, e) {
...
     // https://tools.ietf.org/html/draft-ietf-jose-json-web-key-33
-    let jwk = webCrypto.exportKey('jwk', keyPair.privateKey);
-    jwk = await promisifyIE11Op(jwk, 'Error exporting RSA key pair.');
-
-    // parse raw ArrayBuffer bytes to jwk/json (WebKit/Safari/IE11 quirk)
+    let jwk = await webCrypto.exportKey('jwk', keyPair.privateKey);
+    // parse raw ArrayBuffer bytes to jwk/json (WebKit/Safari quirk)
     if (jwk instanceof ArrayBuffer) {
       jwk = JSON.parse(String.fromCharCode.apply(null, new Uint8Array(jwk)));
     }
```
Comment (twiss) [inline src/crypto/public_key/rsa.js:182]: Do we still need this?

File: src/crypto/public_key/rsa.js [hunk 3]
```diff
@@ -261,15 +225,23 @@ export async function generate(bits, e) {
...
+    n = p.mul(q);
+  } while (n.bitLength() !== bits);
+
+  const phi = p.dec().imul(q.dec());
 
   if (q.lt(p)) {
     [p, q] = [q, p];
   }
-  const phi = p.dec().imul(q.dec());
+
```
Comment (twiss) [inline src/crypto/public_key/rsa.js:242]: I get that it doesn't matter, but I think it was clearer the other way around - then it's sort of more obvious that flipping p and q doesn't cause any issues.

Comment (larabr) [reply src/crypto/public_key/rsa.js:242]: I had the opposite impression, namely that doing it afterwards implied the order mattered 😄  and in the WebCrypto code above we also swap the numbers when returning, so it's consistent with that.

Comment (twiss) [reply src/crypto/public_key/rsa.js:242]: OK, fair enough :+1:

File: src/crypto/public_key/rsa.js (hunk 2 above)
Comment (larabr) [reply src/crypto/public_key/rsa.js:182]: I guess not, I cannot find any docs related to it, but Safari 11 works fine without it

Comment (twiss) [reply src/crypto/public_key/rsa.js:182]: :+1: I guess it was only necessary for `webkitSubtle`, then

**Review by twiss — APPROVED** [review]

**Pull request merged**